CRM/Core/BAO/WordReplacement.php

```diff
@@ -303,13 +303,21 @@ public static function getLocaleCustomStrings($locale, $domainId = NULL) {
    * @return array|mixed
    */
   private static function _getLocaleCustomStrings($domainId) {
-    // TODO: Would it be worthwhile using memcache here?
-    $domain = CRM_Core_DAO::executeQuery('SELECT locale_custom_strings FROM civicrm_domain WHERE id = %1', array(
-      1 => array($domainId, 'Integer'),
-    ));
-    while ($domain->fetch()) {
-      return empty($domain->locale_custom_strings) ? array() : unserialize($domain->locale_custom_strings);
+    $cacheString = __FUNCTION__ . 'locale_custom_strings_' . $domainId;
+    $cached = CRM_Utils_Cache::singleton()->get($cacheString);
+    if (!$cached) {
+      $domainStrings = CRM_Core_DAO::singleValueQuery('SELECT locale_custom_strings FROM civicrm_domain WHERE id = %1', array(
+        1 => array($domainId, 'Integer'),
+      ));
+      if (empty($domainStrings)) {
+        $cached = array();
+      }
+      else {
+        $cached = unserialize($domainStrings);
+      }
+      CRM_Utils_Cache::singleton()->set($cacheString, $cached);
     }
+    return ($cached) ? array() : $cached;
```

Comment:
The funny thing is that this data is already supposed to be cached (`CRM_Core_I18n::crm_translate_raw` calls `getLocaleCustomStrings` and stores it in `Civi::$statics`). But it looks like that caching wasn't working for blank locales.

This patch fixes that other cache and is simpler: https://github.com/civicrm/civicrm-core/pull/7214